images/macos/scripts/build/install-codeql-bundle.sh

```diff
@@ -6,9 +6,24 @@
 
 source ~/utils/utils.sh
 
+# Retrieve the latest major version of the CodeQL Action to use in the base URL for downloading the bundle.
+releases=$(curl -s "https://api.github.com/repos/github/codeql-action/releases")
```

Comment:
This needs to use a PAT [like we do here](https://github.com/actions/runner-images/blob/79ad381addf50d2f8cc28e998e27d25894cac64f/images/macos/scripts/build/install-ruby.sh#L31) or image generation could be throttled and fail

Reply:
Thanks! Previously we already used `curl` on Ubuntu or `download_with_retry` on macOS for another URL without the PAT (see line 25 in this file). Does `download_with_retry` help prevent the throttling?

Also, is this specific to macOS or do you recommend I use the PAT in the corresponding [Ubuntu](https://github.com/actions/runner-images/pull/11931) & [Windows](https://github.com/actions/runner-images/pull/11947) PRs?

Reply:
MacOS image generation has a higher risk of throttling so we have to use PATs. It is less of a problem for Windows/Linux images. You may build Url with this code https://github.com/actions/runner-images/blob/79ad381addf50d2f8cc28e998e27d25894cac64f/images/macos/scripts/helpers/utils.sh#L151